src/app/services/sms/sms.controller.ts

```diff
@@ -0,0 +1,72 @@
+import { StatusCodes } from 'http-status-codes'
+import { Types } from 'mongoose'
+
+import { ErrorDto } from '../../../types/api'
+import { createLoggerWithLabel } from '../../config/logger'
+import { withUserAuthentication } from '../../modules/auth/auth.middlewares'
+import { ControllerHandler } from '../../modules/core/core.types'
+import * as FormService from '../../modules/form/form.service'
+import * as SmsService from '../../services/sms/sms.service'
+import { createReqMeta } from '../../utils/request'
+
+import { SmsCountsMeta } from './sms.types'
+import { mapRouteError } from './sms.util'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Private handler to retrieve the free sms counts remaining for a user and a form belonging to the user
```

Comment:
nit: should it not be the free sms counts remaining for a form by querying counts remaining for the admin?

Reply:
changed because your description is better